### PR TITLE
Add clippy --all-features flag to check non-default crates

### PR DESCRIPTION
**Pull request by mariaschett**

### CODE REVIEW

File: oak_client/src/attestation.rs [hunk 1]
```diff
@@ -84,7 +84,7 @@ pub fn create_tls_attestation_config(
     config.set_protocols(&[b"http/1.1".to_vec(), b"h2".to_vec()]);
 
     // Add root TLS certificate.
-    let mut cc_reader = std::io::BufReader::new(&root_tls_certificate[..]);
+    let mut cc_reader = std::io::BufReader::new(root_tls_certificate);
```
Comment (rbehjati) [inline oak_client/src/attestation.rs:87]: Out of curiosity, what was the linting error for this?

File: oak_loader/src/attestation.rs [hunk 2]
```diff
@@ -50,7 +50,7 @@ pub async fn get_tls_identity_from_proxy(
         "Sending certificate signing request to Proxy Attestation Service: {:?}",
         uri
     );
-    let mut client = create_proxy_client(&uri, &root_tls_certificate)
+    let mut client = create_proxy_client(uri, root_tls_certificate)
```
Comment (rbehjati) [inline oak_loader/src/attestation.rs:53]: What was the error in this case? (if you remember :))

**Review by rbehjati — APPROVED** [review]
Nice. Thanks :)

File: oak_client/src/attestation.rs (hunk 1 above)
Comment (mariaschett) [reply oak_client/src/attestation.rs:87]: The error was:

use the original value instead: `root_tls_certificate` 
= note: `-D clippy::redundant-slicing` implied by `-D warnings`

File: oak_loader/src/attestation.rs (hunk 2 above)
Comment (mariaschett) [reply oak_loader/src/attestation.rs:53]: The error was: "this expression borrows a reference (`&http::Uri` / `&[u8]`) that is immediately dereferenced by the compiler"

**Pull request merged**